js/stringProcessing/countSyllables.js

```diff
@@ -1,133 +1,99 @@
+
+var languageSyllableRegexMaster = new LanguageSyllableRegexMaster( syllableMatchers() );
 
 /**
- * Checks the textstring for exclusion words. If they are found, returns the number of syllables these have, since
- * they are incorrectly detected with the syllablecounters based on regexes.
+ * Counts the syllables by splitting on consonants, leaving groups of vowels.
  *
- * @param {string} text The text to look for exclusionwords
- * @returns {number} The number of syllables found in the exclusionwords
+ * @param {string} word A text with words to count syllables.
+ * @returns {number} the syllable count
  */
-var countExclusionSyllables = function( text ) {
-	var count = 0, matches;
-
-	forEach( exclusionWordsRegexes, function( exclusionWordRegex ) {
-		matches = text.match( exclusionWordRegex.regex );
-
-		if ( matches !== null ) {
-			count += ( matches.length * exclusionWordRegex.syllables );
-		}
+var countUsingVowels = function( word ) {
+	var numberOfSyllables = 0;
+	var foundVowels = word.split( vowelRegex );
+	var filteredWords = filter( foundVowels, function( vowel ) {
+		return vowel !== "";
 	} );
+	numberOfSyllables += filteredWords.length;
 
-	return count;
+	return numberOfSyllables;
 };
 
 /**
- * Removes words from the text that are in the exclusion array. These words are counted
- * incorrectly in the syllable counters, so they are removed and checked sperately.
+ * Counts the syllables using vowel exclusions. These are used for groups of vowels that are more or less
+ * then 1 syllable.
  *
- * @param {string} text The text to remove words from
- * @returns {string} The text with the exclusionwords removed
+ * @param {String} word The word to count syllables in.
+ * @returns {number} The number of syllables found in the given word.
  */
-var removeExclusionWords = function( text ) {
-	forEach( exclusionWordsRegexes, function( exclusionWordRegex ) {
-		text = text.replace( exclusionWordRegex.regex, "" );
-	} );
-
-	return text;
+var countVowelExclusions = function( word ) {
+	return languageSyllableRegexMaster.countSyllables( word );
 };
 
 /**
- * Counts the syllables by splitting on consonants.
+ * Checks if the word is an exclusion word.
  *
- * @param {string} text A text with words to count syllables.
- * @returns {number} the syllable count
+ * @param {String} word The word to check against exclusion words.
+ * @returns {number} The number of syllables found.
  */
-var countBasicSyllables = function( text ) {
-	var array = text.split( " " );
-	var i, j, splitWord, count = 0;
-
-	// split textstring to individual words
-	for ( i = 0; i < array.length; i++ ) {
-
-		// split on consonants
-		splitWord = array[ i ].split( /[^aeiouy]/g );
-
-		// if the string isn't empty, a consonant was found, up the counter
-		for ( j = 0; j < splitWord.length; j++ ) {
-			if ( splitWord[ j ] !== "" ) {
-				count++;
-			}
+var countExclusions = function( word ) {
```

Comment:
But you are not counting the words. You're counting the syllables in these words.